workbench-core/accounts/src/services/projectService.ts

```diff
@@ -263,6 +254,83 @@ export default class ProjectService {
     return newProject;
   }
 
+  /**
+   * Soft deletes a project from the database.
+   *
+   * @param request - a {@link DeleteProjectRequest} object that contains the id of the project to delete
+   * @param checkDependencies - an async function that checks if there are dependencies associated with the project
+   */
+  public async softDeleteProject(
+    request: DeleteProjectRequest,
+    checkDependencies: (projectId: string) => Promise<void>
+  ): Promise<void> {
+    // verify all dependencies are empty
+    await checkDependencies(request.projectId);
+
+    // verify project exists
+    await this.getProject({ projectId: request.projectId });
+
+    // Delete Permissions for the groups--TODO implement after dynamic AuthZ
+    // const identityPermissions: IdentityPermission[] = this._generateIdentityPermissionsForProject(projectId);
+    // const deleteIdentityPermissionsResponse = this._dynamicAuthorizationService.deleteIdentityPermissions(
+    //   identityPermissions
+    // );
+    // if (!createIdentityPermissionsResponse.deleted) {
+    //   throw Boom.badImplementation(
+    //     'Failed to delete batch identity permissions for project with dyamic authorization service.'
+    //   );
+    // }
+
+    // Delete ProjectAdmin and Researcher groups--TODO implement after dynamic AuthZ
+    // const deleteProjectAdminGroupResponse = this._dynamicAuthorizationService.deleteGroup(user, {
+    //   groupId: `${projId}#PA`
+    // });
+    // const deleteResearcherGroupResponse = this._dynamicAuthorizationService.deleteGroup(user, {
+    //   groupId: `${projId}#Researcher`
+    // });
+    // if (!deleteProjectAdminGroupResponse.created || !deleteResearcherGroupResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to delete Project Admin group or Researcher group with dynamic authorization service.'
+    //   );
+    // }
+
+    // delete from DDB
+    const response = await this._aws.helpers.ddb.updateExecuteAndFormat({
```

Comment:
Thoughts on putting this in a try, catch like [here](https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/workbench-core/accounts/src/services/costCenterService.ts#L52). In the catch statement we should `console.error('Failed to delete project ${projectId}}`, e)` so we  have error logs.

Reply:
Thanks for the suggestion! I will change to use that!